src/main/java/com/team4/testingsystem/services/impl/TimerServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package com.team4.testingsystem.services.impl;
+
+import com.team4.testingsystem.entities.Test;
+import com.team4.testingsystem.entities.Timer;
+import com.team4.testingsystem.enums.Status;
+import com.team4.testingsystem.repositories.TimerRepository;
+import com.team4.testingsystem.services.TimerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.TimerTask;
+
+@Service
+public class TimerServiceImpl implements TimerService {
+    private final TimerRepository timerRepository;
+    private final ApplicationEventPublisher eventPublisher;
+
+    @Autowired
+    public TimerServiceImpl(TimerRepository timerRepository,
+                            ApplicationEventPublisher eventPublisher) {
+        this.timerRepository = timerRepository;
+        this.eventPublisher = eventPublisher;
```

Comment:
Allargsconstructor

Reply:
Maybe we should then use lombok for all services?

Reply:
We have a task for it